File: for-local-test/160.Intersection of Two Linked Lists.java
import java.util.Set;
import java.util.LinkedList;

class ListNode {
    int val;
    ListNode next;
    ListNode(int x) {
        val = x;
        next = null;
    }
}

class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        int lenA = length(headA);
        int lenB = length(headB);
        if (lenA > lenB){
            for(int i = 0; i < lenA - lenB; i++){
                headA = headA.next;
            }
        }else{
            for(int i = 0; i < lenB - lenA; i++){
                headB = headB.next;
            }
        }
        while (headA != headB){
            headA = headA.next;
            headB = headB.next;
        }
        return headA;
    }
    public int length(ListNode head) {
        int len = 0;
        while (head != null) {
            len++;
            head = head.next;
        }
        return len;
    }

    public static void main(String[] args) {
        ListNode headA = new ListNode(1);
        headA.next = new ListNode(2);
        headA.next.next = new ListNode(3);
        ListNode headB = new ListNode(4);
        headB.next = headA.next;
        Solution solution = new Solution();
        ListNode intersection = solution.getIntersectionNode(headA, headB);
        if (intersection != null) {
            System.out.println("Intersection at node with value: " + intersection.val);
        } else {
            System.out.println("No intersection.");
        }
    }
}

File: for-local-test/python/160.Intersection of Two Linked Lists.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None
class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> ListNode:

        def length(node: ListNode) -> int:
            len = 0
            while node:
                len += 1
                node = node.next
            return len
        lenA = length(headA)
        lenB = length(headB)
        if lenA > lenB:
            for _ in range(lenA - lenB):
                headA = headA.next
        else:
            for _ in range(lenB - lenA):
                headB = headB.next
        while headA is not headB:
            headA = headA.next
            headB = headB.next
        return headA
    
if __name__ == "__main__":
    # Create two linked lists that intersect
    common = ListNode(8)
    common.next = ListNode(10)

    headA = ListNode(3)
    headA.next = ListNode(7)
    headA.next.next = common

    headB = ListNode(99)
    headB.next = ListNode(1)
    headB.next.next = common

    solution = Solution()
    intersection_node = solution.getIntersectionNode(headA, headB)
    if intersection_node:
        print(f"Intersection at node with value: {intersection_node.val}")  # Output: Intersection at node with value: 8
    else:
        print("No intersection found.")